caom2repo/caom2repo/core.py

```diff
@@ -257,15 +259,15 @@ def put_observation(self, observation):
         """
         assert observation.collection is not None
```

Comment:
Just a question about your use of 'assert' here.  Assert is meant as a development process tool to ensure that particular states are not occurring during the development process, to catch errors and help correct code.  When run in the real-world the assert statements are stripped out during optimization.  So, you have these statements that are their to help development process but get stripped out in production so users don't get slowed down by your code guidance.  Just checking that you are aware of that and that, in production, this asset is not needed.